### DISCUSSION TITLE
Speed differences when saving temp files or only image files

### DISCUSSION THREAD

**Discussion by stefankinell** [karttapullautin/karttapullautin#19] [General]
I did a few test runs yesterday on Linux. My normal set up is to save all the temp files through `savetempfiles=1` since my main work process is to use vectorized data. But yesterdays first test run was with standard .ini-file so I forgott to change that. This resulted in processing the 100 laz-files in about 50 minutes, with 24 threads in batch mode. Super fast!

I then went back and switched the save temp files on. And did a re-run. Now it took about 1:50.

That would mean that saving temp files doubles the processing time. 

@rphlo - you know the program better. Do you think this is reasonable? I looked at the process starting at line 1818 in the perl script and I understand it does some cropping of the files when copying them. Perhaps this is a potential next area to translate to rust to make it go faster?

**Comment by rphlo** [comment]
I agree, the savetempfiles routine from perl script seems outdated, and 1 hours differences doesn't make sense, I can look at that next.

**Comment by rphlo** [comment]
Give a try to latest commit!

**Comment by stefankinell** [comment]
Tried it. First observation on Linux on my i9 with the 100 laz files
Original KP on windows: 5 hours
Last Linux before crop-translation: about 1:50 hours
Latest comitt: 57 minutes 9 seconds. 

:-D

**Comment by stefankinell** [comment]
right now testing on the mac and my smaller pc as well (but under linux)

Nice improvement! Now I am down to 20% of original time when i do batch processing.

**Comment by stefankinell** [comment]
Strange, on Linux I get a large improvement, as listed above. On the Mac I get exactly the same time as before the crop with rust was introduced.

I looked twice but and it looks like the new script contains `rusty-pullauta polylinedxfcrop` commands. 

I will rerun tomorrow, probably do a clean download and see what happens.

**Reply by stefankinell** [reply]
I did a few more test runs today on the Mac. The strange thing is that it seems to be on par with the earlier committ after the blocks/building being translated. It does not seem to be that big of a difference on the Mac for the cropping of the files - perl vs rust. Much more noticeable on Linux.

When I skipped the save files I saved 20 minutes on what was a 1:35 execution time. The equivalent saving on the linux was as a bove about half on a 1:50 execution.

**Reply by rphlo** [reply]
There was still one dxf file cropping that was done in perl, From now on all cropping happen in rust.
